### PR TITLE
Allow individual organisations' user list to be sortable

### LINKED ISSUE #1483: Add Sorting For Organisations
Currently the organisation user list is only sorted by the point count. 

image

### PR DESCRIPTION
**Pull request by slightlyskepticalpotat**
Closes #1483.

### CODE REVIEW

File: judge/views/organization.py [hunk 1]
```diff
@@ -85,18 +85,22 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class OrganizationUsers(OrganizationDetailView):
+class OrganizationUsers(QueryStringSortMixin, OrganizationDetailView):
     template_name = 'organization/users.html'
+    all_sorts = frozenset(('points', 'problem_count', 'rating', 'performance_points', 'user__username'))
```
Comment (TheAvidDev) [inline judge/views/organization.py:90]: Allowing username sorting doesn't fall in line with user list sorting elsewhere. Not sure if we should deviate from that for organizations.

Comment (ComputerGenius152) [reply judge/views/organization.py:90]: I currently have another previous pr [open](https://github.com/DMOJ/online-judge/pull/1523) to ensure consistency, once that is resolved I'll make this consistent with it and un-mark this as a draft.

Comment (ComputerGenius152) [reply judge/views/organization.py]: Username sorting is no longer allowed.

File: judge/views/organization.py [hunk 2]
```diff
@@ -85,18 +85,22 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class OrganizationUsers(OrganizationDetailView):
+class OrganizationUsers(QueryStringSortMixin, OrganizationDetailView):
     template_name = 'organization/users.html'
+    all_sorts = frozenset(('points', 'problem_count', 'rating', 'performance_points'))
```
Comment (kiritofeng) [inline judge/views/organization.py:90]: Is it even possible to sort by points from the UI?

Comment (ComputerGenius152) [reply judge/views/organization.py:90]: I wasn't sure what it does and it's present on the main user list, so I felt leaving it in would be wise.

**Review by kiritofeng — CHANGES_REQUESTED** [review]
Are the headings still properly styled? (I.e. do they still appear white, or are they recoloured into the default color of `a` tags?)

**Comment by slightlyskepticalpotat** [comment]
I think it worked when I tested it last year, but I can retest this in a few days.

File: judge/views/organization.py [hunk 3]
```diff
@@ -85,18 +85,22 @@ def get_context_data(self, **kwargs):
...
+    default_desc = all_sorts
+    default_sort = '-performance_points'
 
     def get_context_data(self, **kwargs):
         context = super(OrganizationUsers, self).get_context_data(**kwargs)
         context['title'] = _('%s Members') % self.object.name
         context['users'] = \
-            ranker(self.object.members.filter(is_unlisted=False).order_by('-performance_points', '-problem_count')
+            ranker(self.object.members.filter(is_unlisted=False).order_by(self.order)
```
Comment (int-y1) [inline judge/views/organization.py:98]: Merge conflict here. Since Nov 2021, `context['users']` uses `users_for_template`.

**Comment by Xyene** [comment]
Superseded by https://github.com/DMOJ/online-judge/pull/197, now merged.

**Pull request closed without merging**